ports/servoshell/desktop/webview.rs

```diff
@@ -785,33 +550,32 @@ where
                             .expect("Thread spawning failed")
                     };
                     if let Err(e) = res {
-                        let reason = format!("Failed to send Prompt response: {}", e);
-                        self.event_queue
-                            .push(EmbedderEvent::SendError(webview_id, reason));
+                        self.send_error(webview_id, format!("Failed to send Prompt response: {e}"))
                     }
                 },
                 EmbedderMsg::AllowUnload(sender) => {
                     // Always allow unload for now.
                     if let Err(e) = sender.send(true) {
-                        let reason = format!("Failed to send AllowUnload response: {}", e);
-                        self.event_queue
-                            .push(EmbedderEvent::SendError(webview_id, reason));
+                        self.send_error(
+                            webview_id,
+                            format!("Failed to send AllowUnload response: {e}"),
+                        )
                     }
                 },
                 EmbedderMsg::AllowNavigationRequest(pipeline_id, _url) => {
                     if let Some(_webview_id) = webview_id {
-                        self.event_queue
-                            .push(EmbedderEvent::AllowNavigationResponse(pipeline_id, true));
+                        servo.allow_navigation_response(pipeline_id, true);
                     }
                 },
                 EmbedderMsg::AllowOpeningWebView(response_chan) => {
-                    // Note: would be a place to handle pop-ups config.
-                    // see Step 7 of #the-rules-for-choosing-a-browsing-context-given-a-browsing-context-name
-                    if let Err(e) = response_chan.send(Some(WebViewId::new())) {
-                        warn!("Failed to send AllowOpeningWebView response: {}", e);
-                    };
+                    let webview = servo.new_auxiliary_webview();
+                    match response_chan.send(Some(webview.id())) {
+                        Ok(()) => self.add(webview),
+                        Err(error) => warn!("Failed to send AllowOpeningWebView response: {error}"),
+                    }
                 },
                 EmbedderMsg::WebViewOpened(new_webview_id) => {
+                    println!("WebView opened: {:?}", new_webview_id);
```

Comment:
Is this meant to be call a `log::debug`? Or is it a statement left over from debugging?

Reply:
This is left over from debugging. We will remove it.